### PR TITLE
Fixes/improve ping logger

### PR DESCRIPTION
**Pull request by itzg**

### CODE REVIEW

File: check/check_http.go [hunk 1]
```diff
@@ -58,7 +58,11 @@ func NewHTTPCheck(base *Base) Check {
 	check := &HTTPCheck{Base: *base}
 	err := json.Unmarshal(*base.RawDetails, &check.Details)
 	if err != nil {
-		log.Error("Error unmarshalling base")
+		log.WithFields(log.Fields{
+			"prefix":  "check_http",
+			"err":     err,
+			"details": string(*base.RawDetails),
+		}).Warn("Unable to unmarshal check details")
```
Comment (rphillips) [inline check/check_http.go]: should we warn here, or error?

File: check/check_ping.go [hunk 2]
```diff
@@ -42,8 +42,9 @@ func NewPingCheck(base *Base) Check {
 	err := json.Unmarshal(*base.RawDetails, &check.Details)
 	if err != nil {
 		log.WithFields(log.Fields{
-			"err":  err,
-			"base": base,
+			"prefix":  "check_ping",
+			"err":     err,
+			"details": string(*base.RawDetails),
 		}).Warn("Unable to unmarshal check details")
```
Comment (rphillips) [inline check/check_ping.go]: same here... error?

File: check/unmarshal.go [hunk 3]
```diff
@@ -56,11 +56,18 @@ func NewCheckParsed(parentContext context.Context, checkIn check.CheckIn) (Check
...
-		return NewHTTPCheck(checkBase), nil
+		return ensureNonNilCheck(NewHTTPCheck(checkBase))
 	case "remote.ping":
-		return NewPingCheck(checkBase), nil
+		return ensureNonNilCheck(NewPingCheck(checkBase))
 	}
 	return nil, errors.New(fmt.Sprintf("Invalid check type: %v", checkBase.CheckType))
 }
+
+func ensureNonNilCheck(ch Check) (Check, error) {
```
Comment (rphillips) [inline check/unmarshal.go]: wdyt, of removing ensureNonNilCheck and putting this logic into resolveCheckType... Feels a little weird.

**Comment by itzg** [comment]
@rphillips we need to wait to merge this until https://github.com/racker/ele/pull/4392 deploys, right?

**Comment by rphillips** [comment]
not necessarily

**Pull request merged**